File: src/StoreCleanArchitecture.Domain/Entities/Product.cs
namespace StoreCleanArchitecture.Domain.Entities;

public class Product
{
    public int? Id { get; set;}
    public required string Name { get; set;}
    public required string Description { get; set; }
    public required string Category { get; set; }
    public double Cost { get; set; }
}

File: src/StoreCleanArchitecture.Web/src/app/models/product.ts
export default class Product{
  id?: number = 0
  name?: string = ''
  description?: string = ''
  category?: string = ''
  cost?: number = 0

  constructor(id?: number, name?: string, description?: string, category?: string, cost?: number){
    this.id = id;
    this.name= name;
    this.description = description;
    this.category = category;
    this.cost = cost;
  }
}
